src/screens/MiningServer.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { t } from 'ttag';
+import {
+  colors,
+} from '../constants';
+import LOCAL_STORE from '../storage';
+import { getGlobalWallet } from "../modules/wallet";
+import { useDispatch, useSelector } from 'react-redux';
+import BackButton from '../components/BackButton';
+import HathorAlert from '../components/HathorAlert';
+import { updateMiningServer } from "../actions";
+
+/**
+ * Screen to change the tx-mining server that the wallet uses to mine transactions
+ *
+ * @memberof Screens
+ */
+function MiningServer() {
+  const wallet = getGlobalWallet();
+  const dispatch  = useDispatch();
+  const currentServer = useSelector(state => state.miningServer);
+
+  /* errorMessage {string} Message to be shown in case of error in form */
+  const [errorMessage, setErrorMessage] = useState('');
+  /* isHardwareWallet {boolean} If the application is using a hardware wallet */
+  const isHardwareWallet = LOCAL_STORE.isHardwareWallet();
+
+  // Declare refs
+  const newServerRef = useRef(null);
+  const pinRef = useRef(null);
+  const alertSuccessRef = useRef();
+  const firstRef = useRef(true);
+
+  useEffect(() => {
+    if (firstRef.current) {
+      firstRef.current = false;
+      return;
+    }
+    alertSuccessRef.current.show(3000);
```

Comment:
Why don't you add this in the `changeServer` method after the `dispatch`? I'm okay continuing with this but I think it would be better.

Reply:
Done